proto/adaptors/balancer/balancer_pool.proto

```diff
@@ -0,0 +1,161 @@
+/*
+* Protos for function calls to the Balancer Pool adaptor.
+*/
+
+syntax = "proto3";
+package steward.v3;
+
+option go_package = "/steward_proto";
+
+import "adaptors/base.proto";
+
+// Represents call data for the Balancer Pool adaptor V1, for managing pool positions on Balancer.
+message BalancerPoolAdaptorV1 {
+    oneof function {
+        /***** BASE ADAPTOR FUNCTIONS *****/
+
+        // Represents function `revokeApproval(ERC20 asset, address spender)`
+        RevokeApproval revoke_approval = 1;
+
+        /***** ADAPTOR-SPECIFIC FUNCTIONS *****/
+
+        // Represents function `relayerJoinPool(ERC20[] tokensIn, uint256[] amountsIn, ERC20 btpOut, bytes[] memory callData)`
+        JoinPool join_pool = 2;
+        // Represents function `relayerExitPool(ERC20 bptIn, uint256 amountIn, ERC20[] memory tokensOut, bytes[] memory callData)`
+        ExitPool exit_pool = 3;
+        // Represents function `stakeBPT(ERC20 _bpt, address _liquidityGauge, uint256 _amountIn)`
+        StakeBPT stake_bpt = 4;
+        // Represents function `unstakeBPT(ERC20 _bpt, address _liquidityGauge, uint256 _amountOut)`
+        UnstakeBPT unstake_bpt = 5;
+        // Represents function `claimRewards(address gauge)`
+        ClaimRewards claim_rewards = 6;
+
+    }
+
+    enum SwapKind {
+        SWAP_KIND_UNSPECIFIED = 0;
+        SWAP_KIND_GIVEN_IN = 1;
+        SWAP_KIND_GIVEN_OUT = 2;
+    }
+
+    // Data for a single swap executed by `swap`. `amount` is either `amountIn` or `amountOut` depending on the `kind` value.
+    message SingleSwap {
```

Comment:
Should we include the message expected types for these submessages like we do for the function call comments?

Reply:
To be clear, I meant in the comment.